File: .skills/openclaw-skills/skills/shifenghu/aicade-galaxy-skill/scripts/setup_env.py
#!/usr/bin/env python3

from pathlib import Path
from typing import Dict


ENV_PATH = Path(".env")

DEFAULTS = {
    "AICADE_GALAXY_BASE_URL": "https://aicadegalaxy.com/agent",
    "AICADE_GALAXY_API_KEY": "xxxx",
    "AICADE_GALAXY_OUTPUT_PATH": "output",
}


def main() -> None:
    existing = load_env(ENV_PATH)

    values = {
        "AICADE_GALAXY_BASE_URL": ask_required(
            "AICADE_GALAXY_BASE_URL",
            existing.get("AICADE_GALAXY_BASE_URL")
            or existing.get("CLAWHUB_BASE_URL")
            or DEFAULTS["AICADE_GALAXY_BASE_URL"],
        ),
        "AICADE_GALAXY_API_KEY": ask_required(
            "AICADE_GALAXY_API_KEY",
            existing.get("AICADE_GALAXY_API_KEY", ""),
            secret=True,
        ),
        "AICADE_GALAXY_OUTPUT_PATH": ask_required(
            "AICADE_GALAXY_OUTPUT_PATH",
            existing.get("AICADE_GALAXY_OUTPUT_PATH")
            or DEFAULTS["AICADE_GALAXY_OUTPUT_PATH"],
        ),
    }

    ENV_PATH.write_text(serialize_env(values), encoding="utf-8")
    print(f"Saved {ENV_PATH}")
    print("Configured header: X-API-Key from AICADE_GALAXY_API_KEY")
    print("Using fixed services path: /admin/gateway/services")
    print(f"Using output directory: {values['AICADE_GALAXY_OUTPUT_PATH']}")
    print(
        f"Output file: {values['AICADE_GALAXY_OUTPUT_PATH']}/aicade-galaxy-skill.json"
    )


def load_env(path: Path) -> Dict[str, str]:
    if not path.exists():
        return {}

    result: Dict[str, str] = {}
    for raw_line in path.read_text(encoding="utf-8").splitlines():
        line = raw_line.strip()
        if not line or line.startswith("#") or "=" not in line:
            continue
        key, value = line.split("=", 1)
        result[key.strip()] = strip_quotes(value.strip())
    return result


def strip_quotes(value: str) -> str:
    if len(value) >= 2 and value[0] == value[-1] and value[0] in {"'", '"'}:
        return value[1:-1]
    return value


def serialize_env(values: Dict[str, str]) -> str:
    return "".join(f"{key}={escape_env_value(value)}\n" for key, value in values.items())


def escape_env_value(value: str) -> str:
    if not value or any(ch.isspace() for ch in value) or '"' in value:
        return '"' + value.replace('"', '\\"') + '"'
    return value


def ask_required(key: str, current: str, secret: bool = False) -> str:
    while True:
        shown = mask_value(current) if current and secret else current
        suffix = f" [{shown}]" if shown else ""
        answer = input(f"{key}{suffix}: ").strip()
        value = answer or current
        if value:
            return value
        print(f"{key} is required.")


def mask_value(value: str) -> str:
    if len(value) <= 4:
        return "*" * len(value)
    return "*" * (len(value) - 4) + value[-4:]


if __name__ == "__main__":
    main()

File: .skills/openclaw-skills/skills/shifenghu/aicade-galaxy-skills/scripts/setup_env.mjs
#!/usr/bin/env node

import { readFile, writeFile } from "node:fs/promises";
import { createInterface } from "node:readline/promises";
import { stdin as input, stdout as output } from "node:process";

const ENV_PATH = ".env";

const DEFAULTS = {
  AICADE_GALAXY_BASE_URL: "https://aicadegalaxy.com/agent",
  AICADE_GALAXY_API_KEY: "xxxx",
  AICADE_GALAXY_OUTPUT_PATH: "output",
};

async function main() {
  const existing = await loadEnvFile(ENV_PATH);
  const rl = createInterface({ input, output });

  try {
    const values = {
      AICADE_GALAXY_BASE_URL: await askRequired(
        rl,
        "AICADE_GALAXY_BASE_URL",
        existing.AICADE_GALAXY_BASE_URL ??
          existing.CLAWHUB_BASE_URL ??
          DEFAULTS.AICADE_GALAXY_BASE_URL,
      ),
      AICADE_GALAXY_API_KEY: await askRequired(
        rl,
        "AICADE_GALAXY_API_KEY",
        existing.AICADE_GALAXY_API_KEY ?? DEFAULTS.AICADE_GALAXY_API_KEY,
        true,
      ),
      AICADE_GALAXY_OUTPUT_PATH: await askRequired(
        rl,
        "AICADE_GALAXY_OUTPUT_PATH",
        existing.AICADE_GALAXY_OUTPUT_PATH ??
          DEFAULTS.AICADE_GALAXY_OUTPUT_PATH,
      ),
    };

    await writeFile(ENV_PATH, serializeEnv(values), "utf8");

    output.write(`Saved ${ENV_PATH}\n`);
    output.write("Configured header: X-API-Key from AICADE_GALAXY_API_KEY\n");
    output.write("Using fixed services path: /admin/gateway/services\n");
    output.write(
      `Using output directory: ${values.AICADE_GALAXY_OUTPUT_PATH}\nOutput file: ${values.AICADE_GALAXY_OUTPUT_PATH}/aicade-galaxy-skill.json\n`,
    );
  } finally {
    rl.close();
  }
}

async function loadEnvFile(path) {
  try {
    const content = await readFile(path, "utf8");
    return parseEnv(content);
  } catch {
    return {};
  }
}

function parseEnv(content) {
  const result = {};

  for (const rawLine of content.split(/\r?\n/)) {
    const line = rawLine.trim();
    if (!line || line.startsWith("#")) {
      continue;
    }

    const equalsIndex = line.indexOf("=");
    if (equalsIndex <= 0) {
      continue;
    }

    const key = line.slice(0, equalsIndex).trim();
    const value = line.slice(equalsIndex + 1).trim();
    result[key] = stripQuotes(value);
  }

  return result;
}

function stripQuotes(value) {
  if (
    (value.startsWith('"') && value.endsWith('"')) ||
    (value.startsWith("'") && value.endsWith("'"))
  ) {
    return value.slice(1, -1);
  }

  return value;
}

function serializeEnv(values) {
  return `${Object.entries(values)
    .map(([key, value]) => `${key}=${escapeEnvValue(value)}`)
    .join("\n")}\n`;
}

function escapeEnvValue(value) {
  if (value === "" || /\s/.test(value) || value.includes('"')) {
    return JSON.stringify(value);
  }

  return value;
}

async function askRequired(rl, key, currentValue, secret = false) {
  while (true) {
    const shownDefault = currentValue ? maskIfNeeded(currentValue, secret) : "";
    const suffix = shownDefault ? ` [${shownDefault}]` : "";
    const answer = await rl.question(`${key}${suffix}: `);
    const value = answer.trim() || currentValue;

    if (value) {
      return value;
    }

    output.write(`${key} is required.\n`);
  }
}

function maskIfNeeded(value, secret) {
  if (!secret) {
    return value;
  }

  if (value.length <= 4) {
    return "*".repeat(value.length);
  }

  return `${"*".repeat(value.length - 4)}${value.slice(-4)}`;
}

await main();
